### PR TITLE
Lots more minor DolphinQt stuffs.

### PR DESCRIPTION
**Pull request by waddlesplash**
Commit messages should be self-explanatory. r? @lioncash, @phire, @JosJuice

### CODE REVIEW

**Comment by waddlesplash** [comment]
Windows buildbot failed because it's on a too-old version of Qt. ping @shuffle2 on that.

Also cc @comex if he wants to look at this.

File: Source/Core/DolphinQt/Host.h [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+#include <string>
+
+enum HostEvent {
+	TitleEvent = QEvent::User + 1,
+};
+
+class HostTitleEvent final : public QEvent
+{
+public:
+	HostTitleEvent(const std::string title);
```
Comment (lioncash) [inline Source/Core/DolphinQt/Host.h]: const reference

File: Source/Core/DolphinQt/MainWindow.cpp [hunk 2]
```diff
@@ -110,9 +115,24 @@ DMainWindow::~DMainWindow()
...
+bool DMainWindow::event(QEvent* e)
+{
+	if (e->type() == HostEvent::TitleEvent)
+	{
+		HostTitleEvent* htev = (HostTitleEvent*)e;
+		m_ui->statusbar->showMessage(QString::fromStdString(htev->m_title), 1500);
+		return true;
+	}
+	else
+		return QMainWindow::event(e);
```
Comment (lioncash) [inline Source/Core/DolphinQt/MainWindow.cpp]: else isn't necessary here

File: Source/Core/DolphinQt/MainWindow.cpp [hunk 3]
```diff
@@ -264,9 +275,15 @@ bool DMainWindow::OnStop()
 	// Ask for confirmation in case the user accidentally clicked Stop / Escape
 	if (SConfig::GetInstance().bConfirmStop)
 	{
-		// Pause emulation
-		Core::SetState(Core::CORE_PAUSE);
-		emit CoreStateChanged(Core::CORE_PAUSE);
+		// Pause emulation if it isn't already
+		bool wasPaused = false;
+		if (Core::GetState() == Core::CORE_PAUSE)
+			wasPaused = true;
```
Comment (lioncash) [inline Source/Core/DolphinQt/MainWindow.cpp]: needs to be braced

**Comment by waddlesplash** [comment]
@lioncash Done, so now this is just waiting for the Qt update.

**Comment by Parlane** [comment]
@dolphin-emu-bot rebuild

**Comment by waddlesplash** [comment]
@dolphin-emu-bot rebuild

**Comment by JosJuice** [comment]
@dolphin-emu-bot rebuild

**Comment by waddlesplash** [comment]
@lioncash LGTU?

**Comment by tony971** [comment]
QTDIR not set or non-existent (pull the submodule?)

- Visual Studio 2015

**Comment by JosJuice** [comment]
@tony971 Did you pull the submodule?

**Comment by tony971** [comment]
Yes.

**Comment by waddlesplash** [comment]
> It builds without this pull.

No, that's just Visual Studio ignoring the DolphinQt build failure. You need the pull in order to build DolphinQt. Also, even if you pulled before, now that the Qt submodule has been updated, you'll have to update your local copy as well.

**Comment by tony971** [comment]
I've got Git set to update sub modules every time I pull.

**Comment by tony971** [comment]
I've got Git set to update sub modules every time I pull.

**Comment by waddlesplash** [comment]
Then you need to be on the latest Git revision, or merge this branch into master. This branch by itself is not enough.

**Comment by tony971** [comment]
I merged this pull request into the latest master after updating the submodule. Everything was up to date and that's why I'm reporting an issue.

**Comment by waddlesplash** [comment]
Please paste the contents of the Externals\Qt directory, as well as the contents of QtCompile.props.

**Comment by tony971** [comment]
I looked in the Externals\Qt directory and got confused why I saw 5.4 stuffs. So I deleted the directory and pulled again. Same result.

https://www.dropbox.com/s/dbzgfwbx17t0zza/Qt.7z?dl=0

**Comment by waddlesplash** [comment]
Sorry, I meant a directory listing, not the actual directory.

Ah, I see -- the submodule is linked to a specific commit, not master. I'll need to fix that.

**Comment by waddlesplash** [comment]
@tony971 Can you pull, update the submodule, and try again?

**Comment by tony971** [comment]
@waddlesplash works for me now :)

**Comment by waddlesplash** [comment]
@lioncash Merge?

**Comment by rukai** [comment]
Wont compile on my machine running Arch Linux.
```
/home/rubic/Projects/dolphin/src/dolphin-dev-waddlesplash/Source/Core/Core/NetPlayServer.cpp: In static member function ‘static bool NetPlayServer::initUPnP()’:
/home/rubic/Projects/dolphin/src/dolphin-dev-waddlesplash/Source/Core/Core/NetPlayServer.cpp:907:74: error: invalid conversion from ‘int*’ to ‘unsigned char’ [-fpermissive]
  UPNPDev *devlist = upnpDiscover(2000, nullptr, nullptr, 0, 0, &upnperror);
                                                                          ^
/home/rubic/Projects/dolphin/src/dolphin-dev-waddlesplash/Source/Core/Core/NetPlayServer.cpp:907:74: error: too few arguments to function ‘UPNPDev* upnpDiscover(int, const char*, const char*, int, int, unsigned char, int*)’
In file included from /home/rubic/Projects/dolphin/src/dolphin-dev-waddlesplash/Source/Core/Core/NetPlayServer.cpp:840:0:
/usr/include/miniupnpc/miniupnpc.h:62:1: note: declared here
 upnpDiscover(int delay, const char * multicastif,
```

It looks like its grabbing miniupnpc from /usr/include instead of from the Externals folder.
I have no idea how your changes have caused that however.

**Comment by BhaaLseN** [comment]
He didn't touch NetPlay, so this PR is not at fault (or should not). Does master work for you?

**Comment by rukai** [comment]
Yes master works fine for me.

I know, its totally bizarre. o.0
I grabbed the zip of dolphin-emu/dolphin and the dolphin-qt branch of waddlesplash/dolphin.
Extracted both, made the build dirs etc.
Compiled them both with `cmake -DENABLE_QT=true ..; make`

dolphin-emu/dolphin works fine
while waddlesplash/dolphin comes up with that error

I would think that the error is more to do with the build process, but that wasnt touched either.
screenshot

**Comment by rukai** [comment]
My bad, nothing is wrong.

I assumed waddlesplash's repo needed to have the full history of the master dolphin in order to be merged, I can now see that isnt the case.
The problem I was experiencing would have been fixed in a recent commit to master, that wasnt in the history of waddlesplash's repo.

**Pull request merged**